src/main/java/core/basesyntax/service/operations/BalanceOperation.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        storageFruits.put(transaction.getFruit(), transaction.getQuantity());
```

Comment:
The storage update logic should ensure that the fruit being added to the storage is not already present with a different quantity. If it is, the quantity should be updated rather than overwritten.